objects/src/accounts/storage/mod.rs

```diff
@@ -33,20 +36,47 @@ impl AccountStorage {
     /// Depth of the storage tree.
     pub const STORAGE_TREE_DEPTH: u8 = 8;
 
+    /// The storage slot at which the slot types commitment is stored.
+    pub const SLOT_TYPES_COMMITMENT_INDEX: u8 = 255;
+
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
     /// Returns a new instance of account storage initialized with the provided items.
-    pub fn new(
-        items: Vec<StorageItem>,
-        store: MerkleStore,
-    ) -> Result<AccountStorage, AccountError> {
-        // construct storage slots smt
-        let slots = SimpleSmt::with_leaves(
-            Self::STORAGE_TREE_DEPTH,
-            items.into_iter().map(|x| (x.0 as u64, x.1)),
-        )
-        .map_err(AccountError::DuplicateStorageItems)?;
-        Ok(Self { slots, store })
+    pub fn new(items: Vec<SlotItem>) -> Result<AccountStorage, AccountError> {
```

Comment:
I wonder if we should simplify this to take items as a set of types `(type, value)` where value is either a slot value or a commitment to the underlying data structure. The main reason for this is that I'm not yet sure what's the right way to handle more complex data types.

For example, we may want to provide data for the complex types as a separate generic auxiliary struct which could be backed either by memory or a database. Or maybe we don't load complex data here at all and leave loading of such data to other places (e.g., transaction executor would load w/e data is needed for transaction execution).

Reply:
With this change we can remove most of the objects that were initially introduced - only the `StorageSlotType` remains. I think this yields a simple data model and cleaner code.